types/ember/index.d.ts

```diff
@@ -431,7 +431,7 @@ declare module 'ember' {
             @param fullName type:name (e.g., 'model:user')
             @param factory (e.g., App.Person)
             **/
-            register(fullName: string, factory: Function, options?: {}): void;
+            register(fullName: string, factory: object, options?: {}): void;
```

Comment:
`.register` is even more flexible than you think -- you can register an `any` (not only `object` as you've indicated). This is the PR to the ember docs, which makes this part of Ember's public API https://github.com/emberjs/ember.js/pull/16801

We should make sure to update all references to this behavior across these types to be consistent. I think we currently have three, and they're all a bit different.

Reply:
I have made another two changes to make sure all `register` to be consistency, but I can't find why tests failed, it seems not related with this PR, any suggestions?